fireworks_poe_bot/plugin.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import Dict, List
+
+
+class LoggingPlugin(ABC):
+    @abstractmethod
+    def log_warn(self, payload: Dict[str, str]):
+        ...
+
+    @abstractmethod
+    def log_info(self, payload: Dict[str, str]):
+        ...
+
+    @abstractmethod
+    def log_error(self, payload: Dict[str, str]):
+        ...
+
+
+_LOGGING_PLUGINS: List[LoggingPlugin] = []
+
+
+def register_logging_plugin(plugin: LoggingPlugin):
+    _LOGGING_PLUGINS.append(plugin)
+
+
+def log_warn(payload: Dict[str, str]):
```

Comment:
Where is the Dict[str, str] restriction coming from? Gcloud logging? Cannot it log things like int values?